www/js/user_page.js

```diff
@@ -1,9 +1,56 @@
+var foodCustom, formChanged, popupOpen, updateUser, user;
+
 $$(document).on('page:init', '.page[data-name="user-page"]', function () {
-  var user = $.auth.user;
+  user = $.auth.user;
+  formChanged = false;
+  foodCustom = user.food_custom;
   initUserPage(user);
 });
 
-function initUserPage(user) {
+const dialog = app.dialog.create({
+  title: 'Osparade ändringar',
+  text: 'Du verkar ha osparade ändringar! Vad vill du göra med dem?',
+  closeByBackdropClick: true,
+  buttons: [
+    {
+      text: 'Släng',
+      onClick: function() {
+        alternativesView.router.back();
+      }
+    },
+    {
+      text: 'Spara',
+      onClick: function() {
+        updateUser(user).then(function() {
+          alternativesView.router.back();
+        });
+      },
+    },
+  ],
+  on: {
+    close: function() {
+      popupOpen = false;
+    },
+    open: function() {
+      popupOpen = true;
+    }
+  }
+});
+
+function backButton() {
+  if (!popupOpen) {
+    // Custom food must be checked separately since it is not part of the form
+    if (formChanged || user.food_custom !== foodCustom) {
+      dialog.open();
+    } else {
+      alternativesView.router.back();
+    }
+  } else {
+    dialog.close();
+  }
+}
+
+function initUserPage() {
   // Fill the page with data that is not in the form like name and profile picture
   fillPageWithNonFromData(user);
```

Comment:
Typo I guess? ;) Shouldn't it be fillPageWithNonFormData (form instead of from)?